sql/core/src/test/scala/org/apache/spark/sql/StreamTest.scala

```diff
@@ -67,7 +68,14 @@ trait StreamTest extends QueryTest with Timeouts {
 
   implicit class RichContinuousQuery(cq: ContinuousQuery) {
     def stopQuietly(): Unit = quietly {
-      cq.stop()
+      try {
+        failAfter(10.seconds) {
+          cq.stop()
+        }
+      } catch {
+        case e: TestFailedDueToTimeoutException =>
+          logError(e.getMessage(), e)
```

Comment:
This is okay to prevent us from blocking other PR, but we should make this actually fail the test on a timeout ASAP.